src/librustc_trans/mir/block.rs

```diff
@@ -314,13 +322,17 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                                                cleanup_bundle.as_ref());
                     fn_ty.apply_attrs_callsite(invokeret);
 
-                    landingpad.at_start(|bcx| for op in args {
-                        self.set_operand_dropped(bcx, op);
+                    landingpad.at_start(|bcx| {
+                        debug_loc.apply_bcx(bcx);
```

Comment:
This method seriously needs a better name. `debug_loc.apply_to(bcx)` or `debug_loc.apply_to_bcx(bcx)`.